morpheus/utils/databricks_utils.py

```diff
@@ -0,0 +1,44 @@
+import json
+import os
+from pathlib import Path
+import logging
+
+logger = logging.getLogger(__name__)
+
+def configure_databricks_connect(databricks_host: str,
+                                  databricks_token: str,
+                                  databricks_cluster_id: str,
+                                  databricks_port: str,
+                                  databricks_org_id: str):
+    if (os.environ.get('DATABRICKS_HOST', None) is None and databricks_host is None):
+        raise Exception("Parameter for databricks host not provided")
+    if (os.environ.get('DATABRICKS_TOKEN', None) is None and databricks_token is None):
+        raise Exception("Parameter for databricks token not provided")
+    if (os.environ.get('DATABRICKS_CLUSTER_ID', None) is None and databricks_cluster_id is None):
+        raise Exception("Parameter for databricks cluster not provided")
+
+    config_file = f"{str(Path.home())}/.databricks-connect"
+    add_config = False
+    
+    config = {
+        "host": os.environ.get('DATABRICKS_HOST', databricks_host),
+        "token": os.environ.get('DATABRICKS_TOKEN', databricks_token),
+        "cluster_id": os.environ.get('DATABRICKS_CLUSTER_ID', databricks_cluster_id),
+        "org_id": databricks_org_id,
+        "port": databricks_port
+    }
```

Comment:
There's a couple of things going on here that we'll want to change.

First, we're prioritizing the host environment variable over the local function parameter. We'll want to switch that so the application can override the host. This:
```python
"host": os.environ.get('DATABRICKS_HOST', databricks_host),
```
becomes this:
```python
"host": os.environ.get('DATABRICKS_HOST') if databricks_host is None else databricks_host,
```

Second, the `None` checks in the first part of the function look a little complicated and also very similar to the line above. Let's try to simplify it and combine the logic at the same time. This:
```python
    if (os.environ.get('DATABRICKS_HOST', None) is None and databricks_host is None):
        raise Exception("Parameter for databricks host not provided")
```
can become:
```python
    if databricks_host is None:
        databricks_host = os.environ.get('DATABRICKS_HOST')
        if databricks_host is None:
            raise Exception("Parameter for databricks_host not provided")
```
Then we're able to remove the logic from the `config = { ... }` and just use the values directly:
```python
    config = {
        "host": databricks_host,
        ...
    }
```

Reply:
Thinking about this a little more, these are the same environment variables used to configure `morpheus/modules/mlflow_model_writer.py`, and it appears to be the only way to configure the model writer.

I'm assuming these may need to be configured independently.

Reply:
@dagardner-nv are you suggesting that there is an existing config/module that needs to be reused?

Reply:
@pthalasta I believe David is suggesting the use of unique environment variable names for this stage specifically because our mlflow implementation already uses (some of) these specific environment variable names. With this in mind, I don't think it makes sense to have the environment variables override the application-specific settings, because the application can declare more than one stage. We will need to rethink how we approach this problem in the general case. For the time being, @pthalasta please update your code according to my [original comment](https://github.com/nv-morpheus/Morpheus/pull/630#discussion_r1315148842), and we can address the issue at a later point.

Reply:
@dagardner-nv Issue created here: https://github.com/nv-morpheus/Morpheus/issues/1174

Reply:
@pthalasta Can you also make sure that the environment variables are prepended with `DELTA_TABLE_`? This will disambiguate them from the mlflow variables.

Reply:
After talking it over with the team, we _don't_ want the `DELTA_TABLE_` prefix. Sorry. :grimacing:

Reply:
marking as resolved as there are no changes needed here